src/databricks/sql/thrift_backend.py

```diff
@@ -1003,6 +1003,10 @@ def cancel_command(self, active_op_handle):
     def handle_to_id(session_handle):
         return session_handle.sessionId.guid
 
+    @staticmethod
+    def handle_to_protocol_version(session_handle):
```

Comment:
Why is this method named this way? The other `handle_to...` methods are converting a session handle from bytes to hex. This method is actually extracting the protocol version, right?

Reply:
@nithinkdb please provide PR description. It's going to help your reviewers to review your PR effectively and quickly.
The PECO ticket itself does not have any description either so this is not good practice.

Reply:
Updated the PR description. Not sure I understand, Jesse. For example handle_to_id just accesses the member variable of sessionId.guid. This is an analogous case no? This is mainly just a good place to put this utility function.

Reply:
I agree it's a good place for it. I think the name is unclear. Elsewhere you call this method `get_session_protocol_version` which is clear ("use the session handle to get the protocol version"). That would be a good / clear name for this method too. Another alternative would be `extract_protocol_version_from_handle`.

Reply:
I disagree.

def get_session_id(self):
        return self.thrift_backend.handle_to_id(self._session_handle)

   def get_session_protocol_version(self):
        return self.thrift_backend.handle_to_protocol_version(self._session_handle)

    def get_session_id_hex(self):
        return self.thrift_backend.handle_to_hex_id(self._session_handle)
      
We currently follow this convention for all of these different functions, why break this convention now? Each of these takes a handle and returns a sub value of it. I'd be fine if we decide to refactor all of these, but i don't think it makes sense to have one outlier.

Reply:
We're not disagreeing here, my friend. Look at the name of the method I'm commenting on: `handle_to_protocol_version`. I'm suggesting changing this name to `get_session_protocol_version` to match what you call it elsewhere (and what similar methods are named) because that indicates what it actually does.

Reply:
A session handle has a name, which is a GUID. That GUID can be expressed in Python `bytes` or as a hexadecimal string. It makes sense to name a method `handle_to_hex_str` because this method clearly indicates that the handle is just being converted to hex.

But `handle_to_protocol_version` doesn't make sense because the handle name doesn't include a protocol version. Rather the session handle object includes a property from which we can fetch the protocol version. So the name makes sense to reflect this. Speaking semantically: the hex string == the byte string. But the hex string != the protocol version.

Reply:
I see.